sentry-sidekiq/spec/sentry/sidekiq/sentry_context_middleware_spec.rb

```diff
@@ -0,0 +1,107 @@
+require "spec_helper"
+
+RSpec.shared_context "sidekiq", shared_context: :metadata do
+  let(:user) { { "id" => rand(10_000) } }
+
+  let(:client) do
+    Sidekiq::Client.new.tap do |client|
+      client.middleware do |chain|
+        chain.add Sentry::Sidekiq::SentryContextClientMiddleware
+      end
+    end
+  end
+
+  let(:random_empty_queue) do
+    Sidekiq::Queue.new(rand(10_000)).tap do |queue|
+      queue.clear
+    end
+  end
+end
+
+RSpec.describe Sentry::Sidekiq::SentryContextServerMiddleware do
+  include_context "sidekiq"
+
+  after { Sidekiq::RetrySet.new.clear }
+
+  it "sets user from the job to events" do
+    perform_basic_setup { |config| config.traces_sample_rate = 1.0 }
+    Sentry.set_user(user)
+
+    queue = random_empty_queue
+    options = { fetch: Sidekiq::BasicFetch.new(queues: [queue.name]) }
+    processor = Sidekiq::Processor.new(nil, options)
```

Comment:
I think the way you initialize sidekiq processor is not compatible with sidekiq v5.2 (which runs in Ruby 2.4 test suite).

instead of initializing the processor directly, can we initialize a manager and get the processor from it? like

https://github.com/getsentry/sentry-ruby/blob/6d2fc60fe21ea3def9da891a470d1a1c6a83760c/sentry-sidekiq/spec/sentry/sidekiq_spec.rb#L15-L19

I will merge this PR once the test is fixed 🙂

Reply:
Fixed!

Reply:
actually it still fails with Sidekiq 5

Reply:
Run tests with sidekiq 5 and confirmed it works.